### PR TITLE
Take care of added records while counting the total number of widgets placed in a dashboard

### PR DESCRIPTION
**Pull request by rtripault**
### What does it do ?

Calculate the real total of dashboard widgets (to "mimic" a rank/placement), instead of the total returned by the server.
### Why is it needed ?

When editing/placing widgets into a dashboard, `Ext.data.Store#getTotalCount()` was used. This method only takes care of "cached" records (as in not counting newly added records -- see the note at the end of description http://docs.sencha.com/extjs/3.4.0/#!/api/Ext.data.Store-method-getTotalCount).

Basically, when editing a dashboard which only had 1 record, every records added/placed later (without save in between) would have a rank of `1`.

Making use of `store.data.length` appears to be more reliable.

### CODE REVIEW

**Comment by rtripault** [comment]
I guess this could have been targeted at master, since that's just a minor "fix"...

File: manager/assets/modext/widgets/system/modx.panel.dashboard.js [hunk 1]
```diff
@@ -324,7 +324,7 @@ Ext.extend(MODx.window.DashboardWidgetPlace,MODx.Window,{
             fld.markInvalid(_('dashboard_widget_err_placed'));
             return false;
         }
-        var rank = s.getTotalCount();
+        var rank = s.data.length;
```
Comment (theboxer) [inline manager/assets/modext/widgets/system/modx.panel.dashboard.js]: @rtripault I think it should be `s.data.length + 1`, otherwise when I add another widget, it has same rank as the previous.

screen shot 2015-06-26 at 15 24 56

Comment (rtripault) [inline manager/assets/modext/widgets/system/modx.panel.dashboard.js]: As far as i remind, ranks should be starting at 0, which should make the `+ 1` useless.
All in all, when ranks are messed up, increment on the number of records is not the solution (we should rather try to find the highest rank then increment that one).

I might triple check that

Comment (theboxer) [inline manager/assets/modext/widgets/system/modx.panel.dashboard.js]: Yeah, maybe grabbing the last row and incrementing it's rank?

**Comment by rtripault** [comment]
PR updated following your advise.
Thanks

**Comment by theboxer** [comment]
@rtripault If you have empty widgets grid, this will not allow you to add a widget in it, because `s.getAt(s.getTotalCount() - 1)` will be undefined

**Comment by rtripault** [comment]
Good call, will take care of that.
Sorry for being so "hasty".

**Comment by rtripault** [comment]
Finally took care of that... good thing my last update was causing issue, since i was reproducing the mistake of checking the store (loaded from ajax) rather than the actual content... while being there, i also did some tweak to fix drag & drop reordering while records were not saved.

Hoping this time things will be fine!

**Comment by theboxer** [comment]
Merged to master & develop
